test/integration/policy_selector_test.go

```diff
@@ -0,0 +1,230 @@
+					{Key: "doesnt-match-anything", Operator: "Exists", Values: []string{}},
+				},
+				status: "Compliant; notification - No objects of kind ConfigMap " +
+					"were matched from the policy objectSelector",
+			},
+		}
+
+		testEntries := []TableEntry{}
+		// Populate the test entries
+		for name, test := range tests {
+			testEntries = append(testEntries, Entry("With objectSelector with "+name, test))
+		}
+
+		// Base test definition
+		baseSelectorTest := func(test selectorTest) {
+			It(policyName+" should be created on the Hub", func(ctx SpecContext) {
+				common.DoCreatePolicyTest(policyYAML, common.GvrConfigurationPolicy)
+
+				By("Patching the objectSelector")
+				selector, err := json.Marshal(test.selector)
+				Expect(err).NotTo(HaveOccurred())
+				jsonPath := "/spec/policy-templates/0/objectDefinition/" +
+					"spec/object-templates/0/objectSelector/matchExpressions"
+				_, err = common.OcHub(
+					"patch", "policies.policy.open-cluster-management.io", policyName,
+					"-n", userNamespace, "--type=json", "-p", `[{
+					"op":"replace", 
+					"path":"`+jsonPath+`",
+					"value": `+string(selector)+`}]`)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			It(policyName+" should have the expected status", func(ctx SpecContext) {
+				Eventually(common.GetLatestStatusMessage(policyName, 0),
+					defaultTimeoutSeconds, 1).Should(Equal(test.status))
+			})
+		}
+
+		///////////////////
+		// Run the tests //
+		///////////////////
+
+		BeforeAll(func(ctx SpecContext) {
+			By("The ConfigMaps should be created on the Managed cluster")
+			By(fmt.Sprintf("Create namespace %s on the Managed cluster", configNamespace))
+			namespace := &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{Name: configNamespace},
+			}
+
+			_, err := clientManaged.CoreV1().Namespaces().Create(ctx, namespace, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			for i, name := range configMapNames {
+				By(fmt.Sprintf("Creating ConfigMap %s/%s on the Managed cluster", configNamespace, name))
+				configMap := &corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:   name,
+						Labels: map[string]string{selectorKey: name},
+					},
+					Data: map[string]string{"name": fmt.Sprintf("%s%d", "testvalue", i)},
+				}
+
+				_, err := clientManaged.CoreV1().ConfigMaps(configNamespace).Create(
+					ctx, configMap, metav1.CreateOptions{})
+				if !k8serrors.IsAlreadyExists(err) {
+					Expect(err).ToNot(HaveOccurred())
+				}
+
+				By(fmt.Sprintf("Creating ConfigMap %s/%s on the Managed cluster", configNamespace, unrelatedConfigMap))
+				_, err = clientManaged.CoreV1().ConfigMaps(configNamespace).Create(ctx, &corev1.ConfigMap{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: unrelatedConfigMap,
+					},
+				}, metav1.CreateOptions{})
+				if !k8serrors.IsAlreadyExists(err) {
+					Expect(err).ToNot(HaveOccurred())
+				}
+			}
+		})
+
+		DescribeTableSubtree("Reporting the correct status", baseSelectorTest, testEntries)
```

Comment:
Is there a reason this isn't just `DescribeTable`?

Also, I think it could be more readable if the `testEntries` were defined directly here, and not converted from a map into a list. Like:
```golang
DescribeTable("Reporting the correct status", baseSelectorTest,
	Entry("With empty objectSelector", defaultTest),
	Entry("With single label exists objectSelector",  selectorTest{
		selector: []v1.LabelSelectorRequirement{
			{Key: selectorKey, Operator: "Exists", Values: []string{}},
		},
		status: "NonCompliant; violation - configmaps [selector-config1, selector-config2] " +
			"found but not as specified in namespace policy-selector-test",
	}),
	...
```

Reply:
> Is there a reason this isn't just `DescribeTable`?

The `DescribeTableSubtree` allows nesting `It` nodes (basically wraps in a `Describe` rather than an `It` as `DescribeTable` does: https://onsi.github.io/ginkgo/#generating-subtree-tables

> Also, I think it could be more readable if the `testEntries` were defined directly here, and not converted from a map into a list.

It's true. Originally I was going to use the test name to generate a unique policy name to allow them to run in parallel, but since I didn't do that I agree it makes more sense to just dump them in as arguments.